### PR TITLE
Add unit test for Date constructor to test basic date creation.

### LINKED ISSUE #19: Add test to Dates.Date
Context
It is best practices to have 100% code coverage.

Details
Add tests for QuantSA.Shared.Dates.Date

### PR DESCRIPTION
**Pull request by rwx-oberholzer-marcelle**
# ✅ Pull Request Checklist

Before submitting your pull request, please ensure the following are complete:

## 🔍 Self Review
- [x] Confirmed that the destination branch is correct.
- [x] I have reviewed my own code carefully.
- [x] I have removed any debugging code or print statements.
- [x] Code changes are scoped to the purpose of this PR (no unrelated changes).
- [x] Comments are clear, and the code is self-explanatory where possible.
- [x] Documentation (if applicable) is updated.

## 📄 Task
- [x] The task/issue title is clear and descriptive.
- [x] The description explains **why** (context) and **what** (details) clearly.
- [x] Related issues are referenced (e.g., "Fixes #123").

## ✅ Code Quality
- [x] There are no new warnings or errors in the build output.
- [x] Code follows the project’s coding standards and formatting rules.

## 🧪 Testing
- [ ] All unit and integration tests pass locally.
- [x] New functionality is covered by tests, or is explicitly marked as not testable.
- [x] Edge cases and error handling have been considered.

## ✅ CI
- [x] CI checks are passing (e.g., build, tests, linting).

## 📄 PR Quality
- [x] The PR title is clear and descriptive.
- [x] The PR description explains **what** and **why** clearly.
- [x] Related issues are referenced (e.g., "Fixes #123").

## 👥 Collaboration
- [x] I have assigned reviewers and/or requested feedback.
- [x] I’ve addressed or responded to all prior review comments (if applicable).

### CODE REVIEW

File: QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        [TestMethod]
+        public void Function_FromYearMonthDay_InitializesCorrectly()
+        {
+            var date = new Date(2023, 12, 25);
+            Assert.AreEqual(2023, date.Year);
+            Assert.AreEqual(12, date.Month);
+            Assert.AreEqual(25, date.Day);
+        }
+
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs:37]: Blank line shouldn't be there.

File: QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+using System;
+
+
+namespace QuantSA.Shared.Test.Dates
+{
+    [TestClass]
+    public class DateTest
+    {
+        [TestMethod]
+        public void Function_FromYearMonthDay_InitializesCorrectly()
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs]: Date_FromYearMonthDay_MembersAreCorrect

File: QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        {
+            var date = new Date(2023, 12, 25);
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs]: If you're going to reuse this, move this to the Setup method for the test class.

**Review by rwx-jacquier-robert — CHANGES_REQUESTED** [review]
Also, add tests till there's 100% coverage, if possible.

**Comment by rwx-oberholzer-marcelle** [comment]
> Also, add tests till there's 100% coverage, if possible.

Managed to change the coverage from 84% to 95%. I added 4 unit tests to 4 methods that showed no test coverage.

File: QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        public void IsLeapYear_ReturnsCorrectly()
+        {
+            Assert.IsTrue(Date.IsLeapYear(2020));
+            Assert.IsFalse(Date.IsLeapYear(2021));
+        }
+
+        [TestMethod]
+        public void DaysInMonth_ReturnsCorrectly()
+        {
+            Assert.AreEqual(31, Date.DaysInMonth(2023,1));
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs]: Spacing
```suggestion
            Assert.AreEqual(31, Date.DaysInMonth(2023, 1));
```

File: QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+        [TestMethod]
+        public void IsEndOfMonth_ReturnsCorrectly()
+        {
+            var lastDay = new Date(2023, 4, 30);
+            var notLastDay = new Date(2023, 4, 29);
+
+            Assert.IsTrue(Date.IsEndOfMonth(lastDay));
+            Assert.IsFalse(Date.IsEndOfMonth(notLastDay));
+        }
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs]: Split this into two tests, one for the lastDay, one for notLastDay

File: QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+            Assert.AreEqual(2023, _date.Year);
+            Assert.AreEqual(12, _date.Month);
+            Assert.AreEqual(25, _date.Day);
+        }
+
+        [TestMethod]
+        public void IsLeapYear_ReturnsCorrectly()
+        {
+            Assert.IsTrue(Date.IsLeapYear(2020));
+            Assert.IsFalse(Date.IsLeapYear(2021));
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs]: Split this into two tests.

**Review by rwx-jacquier-robert — CHANGES_REQUESTED** [review]

File: QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        [TestMethod]
+        public void IsEndOfMonth_ReturnsTrueCorrectly()
+        {
+            var lastDay = new Date(2023, 4, 30);
+
+            Assert.IsTrue(Date.IsEndOfMonth(lastDay));
+        }
+
+        [TestMethod]
+        public void IsEndOfMonth_ReturnsFalseCorrectly()
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs]: Test naming is still an issue, for instance, this one should be "IsEndOfMonth_29Apr_ReturnsFalse".

That way you're specifying the state (29Apr) which is never an end of month, and the "correctly" part at the end, which is redundant, is removed.

**Review by rwx-jacquier-robert — CHANGES_REQUESTED** [review]
Look at the one comment I added, revisit all the test names and refactor them.

File: QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs (hunk 7 above)
Comment (rwx-oberholzer-marcelle) [reply QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs]: I will rename all of the tests, thanks Rob!

**Review by rwx-jacquier-robert — CHANGES_REQUESTED** [review]
Revisit the test method names as per the other PRs.

File: QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs [hunk 8]
```diff
@@ -0,0 +1,74 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using QuantSA.Shared.Dates;
+using System;
+
+namespace QuantSA.Shared.Test.Dates
+{
+    [TestClass]
+    public class DateTest
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs]: class name should match file name "DateTests"

File: QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+        [TestMethod]
+        public void DaysInMonth_ReturnsAsExpected()
+        {
+            Assert.AreEqual(31, Date.DaysInMonth(2023, 1));
+            Assert.AreEqual(29, Date.DaysInMonth(2020, 2));
+            Assert.AreEqual(28, Date.DaysInMonth(2023, 2));
+        }
+
+        [TestMethod]
+        public void EndOfMonth_WithMidMonthDate_ReturnsAsExpected()
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Shared.Test/Dates/DateTests.cs]: ReturnsExpectedResult is a better suffix.

File: QuantSA/QuantSA.Shared.Test/QuantSA.Shared.Test.csproj [hunk 10]
```diff
@@ -7,9 +7,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.5.0" />
-    <PackageReference Include="MSTest.TestAdapter" Version="2.1.2" />
-    <PackageReference Include="MSTest.TestFramework" Version="2.1.2" />
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Shared.Test/QuantSA.Shared.Test.csproj:12]: Where are these changes coming from?

**Review by rwx-jacquier-robert — CHANGES_REQUESTED** [review]

File: QuantSA/QuantSA.Shared.Test/QuantSA.Shared.Test.csproj (hunk 10 above)
Comment (rwx-oberholzer-marcelle) [reply QuantSA/QuantSA.Shared.Test/QuantSA.Shared.Test.csproj:12]: @robertjacquier-rw 
I made the changes to update the test packages to newer versions. I was running into some issues with the tests and saw that the we had were old.
